### PR TITLE
Fixed #838, #860 and unnamed deserializers

### LINKED ISSUE #838: RecipeAlternatesWidget -> RecipeAlternativesWidget
Alternative seems to be a more common variant of the word. "Alternates" makes me think that it renders a recipe with alternating items (like those shown by JEI when several item variants are accepted).

### PR DESCRIPTION
**Pull request by YanisBft**
- #838 - Renamed all "alternate" to "alternative"
- #860 - Renamed `stats` field to `statHandler` and `getStats` method to `getStatHandler` in `ClientPlayerEntity` so they match their equivalent in `ServerPlayerEntity`

### CODE REVIEW

**Review by Juuxel — APPROVED** [review]

File: mappings/net/minecraft/world/gen/feature/MegaTreeFeatureConfig.mapping [hunk 1]
```diff
@@ -4,6 +4,7 @@ CLASS net/minecraft/class_4636 net/minecraft/world/gen/feature/MegaTreeFeatureCo
 	METHOD <init> (Lnet/minecraft/class_4651;Lnet/minecraft/class_4651;Ljava/util/List;III)V
 		ARG 5 heightInterval
 		ARG 6 crownHeight
+	METHOD method_23408 deserialize2 (Lcom/mojang/datafixers/Dynamic;)Lnet/minecraft/class_4636;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/feature/MegaTreeFeatureConfig.mapping]: Hmm, is there a name conflict?

**Review by liach — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/world/gen/feature/MegaTreeFeatureConfig.mapping (hunk 1 above)
Comment (YanisBft) [reply mappings/net/minecraft/world/gen/feature/MegaTreeFeatureConfig.mapping]: Yeah, the parent class already has a `deserialize` method. Tbh, I'm not a big fan of `deserialize2`, but it's already used in BranchedTreeFeatureConfig and I don't have any other idea

Comment (liach) [reply mappings/net/minecraft/world/gen/feature/MegaTreeFeatureConfig.mapping]: if both methods are static, then it's an enigma bug.  try rename the method in a text editor and see if travis builds.

Comment (Juuxel) [reply mappings/net/minecraft/world/gen/feature/MegaTreeFeatureConfig.mapping]: Yeah, I'm pretty sure it's an Enigma bug. I encountered it with #858 but forgot to make an issue.

Comment (YanisBft) [reply mappings/net/minecraft/world/gen/feature/MegaTreeFeatureConfig.mapping]: Done, and travis builds

**Review by liach — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**